common/src/IO/AssimpParser.cpp

```diff
@@ -327,6 +343,126 @@ std::optional<size_t> getMeshIndex(const aiScene& scene, const aiMesh& mesh)
   return std::nullopt;
 }
 
+/**
+ * Gets the channel index for a particular assimp node, matched by name.
+ */
+std::optional<size_t> getChannelIndex(const aiAnimation& animation, const aiNode& node)
+{
+  for (unsigned int i = 0; i < animation.mNumChannels; ++i)
+  {
+    const aiNodeAnim* ch = animation.mChannels[i];
```

Comment:
Please use `auto`, plus I would prefer a reference for everything that cannot be null, so `const auto& ch = *animation.mChannels[i]`.